app/controllers/users/users.profile.server.controller.js

```diff
@@ -54,3 +54,66 @@ exports.update = function(req, res) {
 exports.me = function(req, res) {
 	res.json(req.user || null);
 };
+
+/**
+ * Record User's vote and update Article vote counts
+ */
+exports.vote = function(req, res) {
+	var article = req.article;
+	var user = req.user;
+	var newVoteDirection = req.params.direction;
+	var oldVoteDirection;
+	var upVotesAddend = 0;
+	var downVotesAddend = 0;
+	var voteIndex;
+
+	// look for existing vote
+	for (voteIndex = 0; voteIndex<user.votes.length; voteIndex++) {
+		if (article._id.equals(user.votes[voteIndex]._id)) {
```

Comment:
You can compare the String values using `.toString()` though, which I think is simpler and more JS style.